conda/common/logic.py

```diff
@@ -314,61 +330,175 @@ def from_name(self, name):
     def from_index(self, m):
         return self.indices.get(m)
 
-    def Assign_(self, vals, name=None):
-        x = self._assign_no_name(vals)
+    def _assign(self, vals, name=None):
+        x = self._clauses.assign(vals)
         if not name:
             return x
         if vals in {TRUE, FALSE}:
-            x = self._new_var()
-            self.add_clause((x,) if vals else (-x,))
+            x = self._clauses.new_var()
+            self._clauses.add_clause((x,) if vals else (-x,))
         return self.name_var(x, name)
 
-    def _assign_no_name(self, vals):
+    def _convert(self, x):
+        if isinstance(x, (tuple, list)):
+            return type(x)(map(self._convert, x))
+        if isinstance(x, int):
+            return self._check_literal(x)
+        name = x
+        try:
+            return self.names[name]
+        except KeyError:
+            raise ValueError("Unregistered SAT variable name: {}".format(name))
+
+    def _eval(self, func, args, polarity, name):
```

Comment:
I dropped the `conv=True` parameter here, meaning now we `_convert` all inputs. That parameter has only been used by `LinearBound` which is temporarily removed by this commit from the `Clauses` interface. `LinearBound` is re-added in https://github.com/conda/conda/pull/9427/commits/4f0f7bc3acf8f321a750eac50006319b23faa265.